src/formatDate.js

```diff
@@ -50,7 +50,85 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [ , , , fromSeparator ] = fromFormat;
+  const [ , , , toSeparator ] = toFormat;
+  let year, smallYear, month, day;
+  const resultArray = [];
+  let resultDate = '';
+
+  const fromArray = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY': {
+        year = fromArray[i];
+
+        break;
+      }
+
+      case 'YY': {
+        smallYear = fromArray[i];
+
+        break;
+      }
+
+      case 'MM': {
+        month = fromArray[i];
+
+        break;
+      }
+
+      case 'DD': {
+        day = fromArray[i];
+
+        break;
+      }
+    }
+  }
+
+  if (year) {
```

Comment:
In this condition, if year is an empty string (''), the condition will return false although it's a valid value. Consider checking against undefined or null.